### PR TITLE
Added skip option.

### PR DESCRIPTION
**Pull request by smykhailov**
It allows to define RegExp to skip validation for files or folders.

### CODE REVIEW

File: src/DocLinkChecker/Program.cs [hunk 1]
```diff
@@ -195,6 +205,17 @@ private static void ProcessFiles(DirectoryInfo folder)
 
             foreach (FileInfo fi in files)
             {
+                if (!string.IsNullOrEmpty(options.SkipPattern))
+                {
+                    message.Verbose($"Skip pattern: {options.SkipPattern}");
+
+                    if (Regex.IsMatch(fi.FullName.ToLowerInvariant(), options.SkipPattern))
```
Comment (Ellerbach) [inline src/DocLinkChecker/Program.cs:212]: why do you compare in lower invariant? If you are using a regex, you can handle this in the regex?

**Comment by mtirionMSFT** [comment]
@smykhailov I like the proposal. Currently I don't think it's the best solution to add it this way, as it might be harder to move forward. Let me explain what I mean ... :) 

Good to know is that I'm currently working on a complete rewrite of this tool. It will make it more standard .NET, better architected, but more important - enhance the feature set while being backwards compatible in use. One of things I'm adding is the option to use a configuration file like docfx.json. Because of your idea, I'll be adding Include and Exclude options for folders as well using the glob pattern (https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing). There is automatic support for includes and excluding there. This is better readable and maintainable I think then writing complex regex expressions, which isn't that easy for most of us (including me :p).

You can expect an updated version of the DocLinkChecker with this functionality in the coming weeks.

File: src/DocLinkChecker/Program.cs (hunk 1 above)
Comment (smykhailov) [reply src/DocLinkChecker/Program.cs]: you are right, removed

**Comment by mtirionMSFT** [comment]
@smykhailov Please have a look at the branch https://github.com/Ellerbach/docfx-companion-tools/tree/doclinkchecker-upgrade with the new implementation. You would need to pull and compile of course, but you can see if it meets your needs. The DocLinkChecker README describes the new option to include a configuration file.

**Comment by Ellerbach** [comment]
@smykhailov feedback welcome as well ;-)

**Comment by smykhailov** [comment]
this is really great ad exactly what we need. Thank you @mtirionMSFT and @Ellerbach

**Comment by mtirionMSFT** [comment]
Solved by #32

**Pull request closed without merging**